### ISSUE TITLE
compare to datapackage in R for similar API

### ISSUE THREAD

**Issue by mbjones** [DataONEorg/d1_pypackage#3]
Design this python implementation to be compatible with the corresponding R [datapackage](https://github.com/ropensci/datapackage) package.  The `datapackage` implementation in R is under development, and has an API that includes classes for `DataObject`, `DataPackage`, `ResourceMap`, and `SystemMetadata`.  It would be nice if there was a correspondence in the APIs.

**Comment by datadavev** [comment]
This project is primarily to support serialization of the package to a local file system such that it may be reliably used by other tools / services. Much of the implementation (e.g. classes for System Metadata, Resource Maps, and Data Object) has been available in the DataONE python libraries for some time - it would be interesting to evaluate the R implementation in that context.

**Comment by mbjones** [comment]
Yeah, in R we decided to make `dataone` depend on `datapackage` (rather than the inverse), so that other non-dataone R code could also use and create data packages for other systems.  This meant that `SystemMetadata` was defined in `datapackage`, and the `dataone` library then imports that and adds all of the service calls.  So R's `datapackage` is a standalone data package representation that can serialize to BagIt with SystemMetadata and does not depend on having access to any other DataONE services.  It would be nice if the BagIt-based package format was standalone on all platforms, even though it means some refactoring. Open to discussion...
